packages/oidc/src/refresh/refreshGrant.ts

```diff
@@ -76,6 +76,12 @@ export async function refresh(
     };
   }
 
+  if (client.clientId === undefined) {
```

Comment:
Let's move this up to line 71-ish, no need to generate DPoP header if we're just gonna throw (though do note that may change from an async throw to a sync throw potentially)

Reply:
Good point, I actually moved it all the way to the top of the function for clarity